File: core/item/Card.ts
import { Sized } from "./Sized.ts";

export enum CardStatus {
  IN = "IN",
  OUT = "OUT",
  LOST = "LOST",
}

export const CARD_TYPES = [
  "Bank card",
  "Driving license",
  "National card",
];

export abstract class Card implements Sized {
  private static readonly SIZE = 1;

  constructor(
    public readonly type: typeof CARD_TYPES[number],
    public readonly owner: string,
  ) {
  }

  computeSize(): number {
    return Card.SIZE;
  }

  toString() {
    return this.type + ":" + this.owner;
  }
}

export class BankCard extends Card {
  constructor(owner: string) {
    super("Bank card", owner);
  }
}

export class DrivingLicense extends Card {
  constructor(owner: string) {
    super("Driving license", owner);
  }
}

export class NationalCard extends Card {
  constructor(owner: string) {
    super("National card", owner);
  }
}

File: core/item/card.py
from core.item.sized import Sized

class Card(Sized):
  def __init__(self, type, owner):
    self.owner = owner
    self.type = type

  def compute_size(self):
    return 1

class BankCard(Card):
  def __init__(self, owner):
    super(owner)

class NationalCard(Card):
  def __init__(self, owner):
    super(owner)

class DrivingLicense(Card):
  def __init__(self, owner):
    super(owner)

def get_card_creator_from_type(type):
  if type == "Bank card":
    return BankCard
  elif type == "National card":
    return NationalCard
  elif type == "Driving license":
    return DrivingLicense